tests/suites/storage/application_storage.sh

```diff
@@ -0,0 +1,71 @@
+run_application_storage_directives() {
+	echo
+
+	model_name="application-storage"
+	file="${TEST_DIR}/test-${model_name}.log"
+
+	ensure "${model_name}" "${file}"
+	app_name="postgresql"
+
+	# Deploy an application with storage directive pgdata=2G.
+	if [ "${BOOTSTRAP_PROVIDER:-}" = "k8s" ]; then
+		echo "Deploying k8s application"
+		app_name="postgresql-k8s"
+		juju deploy "$app_name" --storage pgdata=2G
+		juju trust "$app_name" --scope=cluster
+	else
+		echo "Deploying non-k8s application"
+		app_name="postgresql"
+		juju deploy "$app_name" --storage pgdata=2G
+	fi
+
+	# Wait for application to be active.
+	wait_for true ".applications[\"$app_name\"][\"application-status\"].current==\"active\""
+
+	juju application-storage "$app_name" --format=yaml
+	old_size=$(juju application-storage "$app_name" --format=json | jq -r '.pgdata.Size')
+	if [ "$old_size" != "2048" ]; then
+		echo "Expected storage size 2048Mi, got $old_size"
+		exit 1
+	fi
+
+	# Update the storage directive pgdata=3G.
+	juju application-storage "$app_name" pgdata=3G
+	new_size=$(juju application-storage "$app_name" --format=json | jq -r '.pgdata.Size')
+	if [ "$new_size" != "3072" ]; then
+		echo "Expected updated storage size to be 3072Mi, got $new_size"
+		exit 1
+	fi
+
+	# Add 2 more units and check their storage size is 3G.
```

Comment:
For k8s can we add a test that the PVC has the expected size. We can query with kubectl etc